lib/recurly/pricing/subscription/calculations.js

```diff
@@ -182,19 +181,23 @@ export default class Calculations {
     this.price.now.discount = 0;
     this.price.next.discount = 0;
 
-    if (coupon) {
-      if (coupon.discount.rate) {
-        var discountNow = parseFloat((this.price.now.subtotal * coupon.discount.rate).toFixed(6));
-        var discountNext = coupon.single_use ? 0 : parseFloat((this.price.next.subtotal * coupon.discount.rate).toFixed(6));
-        this.price.now.discount = Math.round(discountNow * 100) / 100;
-        this.price.next.discount = Math.round(discountNext * 100) / 100;
-      } else if (coupon.discount.type === 'free_trial') {
-        // Handled in separate trial logic
-      } else {
-        this.price.now.discount = coupon.discount.amount[this.items.currency];
-        this.price.next.discount = coupon.single_use ? 0 : coupon.discount.amount[this.items.currency];
-      }
+    if (!coupon) return;
+
+    if (coupon.discount.rate) {
+      var discountNow = parseFloat((this.price.now.subtotal * coupon.discount.rate).toFixed(6));
+      var discountNext = parseFloat((this.price.next.subtotal * coupon.discount.rate).toFixed(6));
+      this.price.now.discount = Math.round(discountNow * 100) / 100;
+      this.price.next.discount = Math.round(discountNext * 100) / 100;
+    } else if (coupon.discount.type === 'free_trial') {
+      // Handled in separate trial logic
+    } else {
+      const discountableNow = this.price.now.subtotal + this.price.now.setup_fee;
+      const discountableNext = this.price.next.subtotal;
+      this.price.now.discount = Math.min(discountableNow, coupon.discount.amount[this.items.currency]);
+      this.price.next.discount = Math.min(discountableNext, coupon.discount.amount[this.items.currency]);
```

Comment:
This is an important change to current pricing behavior.

Previously, if a gift card with a value higher than the subscription total (now + next cycle) was applied, it would apply a discount larger than the total. Now, the discount amount is limited to the total per cycle.